en/services/mission.md

```diff
@@ -297,9 +307,59 @@ The protocol has been implemented in C.
 Source code:
 * [src/MissionManager/PlanManager.cc](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/PlanManager.cc)
 
+
 ### ArduPilot
 
-TBD
+ArduPilot implements the mission protocol in C.
+
+ArduPilot uses the same messages and message flow described in this specification. 
+There are some implementation diferences that affect compatibility.
+These are documented below.
+
+Source:
+* [/libraries/GCS_MAVLink/GCS_Common.cpp](https://github.com/ArduPilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp)
+
+
+#### Flight Plan Missions
+
+Mission upload, download, clearing missions, and monitoring progress and partial mission upload ([MISSION_WRITE_PARTIAL_LIST](#MISSION_WRITE_PARTIAL_LIST)) are supported.
+
+[Partial mission download](#download_partial) is not supported ([MISSION_REQUEST_PARTIAL_LIST](#MISSION_REQUEST_PARTIAL_LIST)).
+
+ArduPilot's implementation differs from this specification (non-exhaustively):
+- The first mission sequence number (`seq==0`) is populated with the home position of the vehicle instead of the first mission item.
+- Mission uploads are not "atomic". 
+  An upload that fails (or is canceled) part-way through will not match the pre-update state.
+  Instead it may be a mix of the original and new mission.
+- Even if upload is successful, the vehicle mission may not match the version on the uploading system.
+  - If you try and upload more items than ArduPilot can store the system will "accept" the items (i.e. not report a failure) but will just overwrite each new item to the same (highest) slot in the mission list.
+  - Only fields that are used are stored; if an uploaded mission is then downloaded it will not survive the round trip unaltered.
```

Comment:
There is also rounding on some fields (and internal max possible value due to storage space). If you try to do a straight comparison of the float params before/after upload this can cause failures.